src/main/java/carracing/car/MaxMoving.java

```diff
@@ -0,0 +1,32 @@
+package carracing.car;
+
+import carracing.controlstatement.Equal;
+import carracing.controlstatement.IF;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MaxMoving implements WinnerStrategy {
+	@Override
+	public List<Car> winners(List<Car> cars) {
+		int maxMovingDistance = maxMovingDistance(cars);
+		List<Car> winners = winnersWithMaxMovingDistance(cars, maxMovingDistance);
+
+		return winners;
+	}
+
+	private int maxMovingDistance(List<Car> cars) {
+		return cars.stream()
+				.map(Car::movingDistance)
+				.max(Comparator.comparingInt(movingDistance -> movingDistance))
+				.orElseThrow(IllegalArgumentException::new);
+	}
+
+	private List<Car> winnersWithMaxMovingDistance(List<Car> cars, int maxMovingDistance) {
+		return cars.stream()
+				.filter(car -> new IF(new Equal(car.movingDistance(), maxMovingDistance)).booleanValue()
```

Comment:
IF는 생략해도 좋을 것 같아요 😃 
객체지향 패러다임에 따라 객체에게 메시지를 보내면 어떨까요?
자동차의 위치에 대해서는 가장 잘 알고 있으니 car 객체에게 메시지를 보낼 수 있겠네요
```suggestion
				.filter(car -> car.sameDistance(maxMovingDistance)
```